apps/api/src/employee/employee.service.ts

```diff
@@ -43,24 +51,83 @@ export class EmployeeService {
     return this.ienUsersRepository.findOne({ email });
   }
 
+  _nameSearchQuery(keyword: string) {
+    let keywords = keyword.split(' ');
+    keywords = keywords.filter(item => item.length);
+    if (keywords.length === 1) {
+      return `(Employee.name ilike '%${keywords[0].toLowerCase()}%')`;
+    } else if (keywords.length === 2) {
+      return `(Employee.name ilike '%${keywords[0]}%${keywords[1]}%' OR Employee.name ilike '%${keywords[1]}%${keywords[0]}%')`;
+    } else if (keywords.length === 3) {
+      const possibleShuffle = [];
+      possibleShuffle.push(`Employee.name ilike '%${keywords[0]}%${keywords[1]}%${keywords[2]}%'`);
```

Comment:
It might be a quite expensive query. I hope we could come up with a better way later.

Reply:
I think this will go hand it hand with how we search names overall, the search uses the same queries in applicant table